puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -38,7 +38,7 @@ def report_to_hash
       end
 
       resources = build_resources_list
-      is_noop = resources.any? { |rs| has_noop_event?(rs) } and resources.none? { |rs| has_failed_event?(rs) }
+      is_noop = resources.any? { |rs| has_noop_event?(rs) } && resources.none? { |rs| has_enforcement_event?(rs) }
```

Comment:
since a resource can have multiple events, we should make sure they're all noop instead of that they each have a noop. Additionally this could be rewritten to check
```
resources.all? { |rs| all_noop_events?(rs) }
```
to avoid a second pass through the list of resources.